### PR TITLE
String literal in returned ternary expression no longer crashes (alliedmodders/sourcepawn#940)

### PR DESCRIPTION
**Pull request by NotnHeavy**
Addresses #940. Hopefully the coding style fits?

### CODE REVIEW

File: compiler/semantics.h [hunk 1]
```diff
@@ -182,6 +182,7 @@ class Semantics final
     bool CheckAssertStmt(AssertStmt* stmt);
     bool CheckStaticAssertStmt(StaticAssertStmt* stmt);
     bool CheckReturnStmt(ReturnStmt* stmt);
+    bool CheckReturnNode(ReturnStmt* stmt, FunctionDecl* fun, Expr* expr) ;
```
[the remark this answers is not in the document]
Comment (NotnHeavy) [reply compiler/semantics.h:185]: i have no idea why i added a space there :(

**Comment by NotnHeavy** [comment]
I'm guessing this means putting test code into `/tests/compile-only/`? This is my first pull request for SourcePawn so I'm not perfect with what's supposed to be done when it comes to submitting changes haha.

**Comment by NotnHeavy** [comment]
Oh the test's failing but I suppose the error's intentional - I'll rename the file so that it's prefixed with `fail` rather than `ok`. Sorry, been really tired today, not sure why...

**Comment by dvander** [comment]
Thanks for the fix. This is on my shortlist of things to look at. In theory, we shouldn't have to special case anything here, as the `value` of the TernaryExpr should be enough. And ReturnExpr with arrays is a degenerate form of assignment and we don't special case TernaryExpr in assignment. But... I could believe this is the right thing to do, given the archaic way arrays are handled.

**Comment by NotnHeavy** [comment]
> Thanks for the fix. This is on my shortlist of things to look at. In theory, we shouldn't have to special case anything here, as the `value` of the TernaryExpr should be enough. And ReturnExpr with arrays is a degenerate form of assignment and we don't special case TernaryExpr in assignment. But... I could believe this is the right thing to do, given the archaic way arrays are handled.

Yeah that's fair, I just thought I'd send this. I don't have much knowledge with the SourcePawn compiler/vm still, nor do I fully understand your intents, but I do wanna tinker with SourcePawn :)

Also my idiot brain forgot to add a .txt for the `fail` file....

**Comment by dvander** [comment]
Closing as this was fixed as part of the enum struct refactoring.

**Pull request closed without merging**